core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -474,10 +474,12 @@ private[spark] class SparkSubmit extends Logging {
         args.mainClass = "org.apache.spark.deploy.PythonRunner"
         args.childArgs = ArrayBuffer(localPrimaryResource, localPyFiles) ++ args.childArgs
       }
-      if (clusterManager != YARN) {
-        // The YARN backend handles python files differently, so don't merge the lists.
-        args.files = mergeFileLists(args.files, args.pyFiles)
-      }
+    }
+
+    // Non-PySpark applications will also need Python dependencies.
+    if (deployMode == CLIENT && clusterManager != YARN) {
```

Comment:
Just a question. Did we `mergeFileLists` for `deployMode != CLIENT` already?

Reply:
No, I don't see we merge it for cluster deploy mode.

Reply:
Got it. Thanks, @viirya .